### DISCUSSION TITLE
search-api health check API throws error "Elastic Search index is not avaialable"

### DISCUSSION THREAD

**Discussion by krishnakr6** [Sunbird-Knowlg/Community#101] [Documentation Issues]
Followed this link  [asses-management-service](https://knowlg.sunbird.org/use/installation-guide/services/asset-management-service) to install.
Below are the more details
**Search-API**
**Request**
curl --location 'http://localhost:9000/health' \ --header 'Content-Type: application/json' 

**Response**
{ "id": "api.search-service.health", "ver": "3.0", "ts": "2023-02-24T16:33:27Z+05:30", "params": { "resmsgid": "80f18e9b-6a93-431a-ab47-72ce8b6c28af", "msgid": null, "err": null, "status": "failed", "errmsg": "Elastic Search index is not avaialable" }, "responseCode": "SERVER_ERROR", "result": { "healthy": false } }

When i debug the error , there was no elastic search installed.Further post installation i was getting error. According to code the health check were looking for compositesearch index in ES.
 
I have noticed below gaps with documentation and the actual required installation  guide.
1. ES setup details is not available
2. ES index creation instructions.

could you let me know how to proceed next with this issue. Further  i will be happy to contribute on this specific issue by providing documentation or code changes if required.

**Comment by aimansharief** [comment]
@krishnakr6 
To fix the issue create index and mappings for **compositesearch** . To get the data to create index and mappings you can refer this 
https://github.com/project-sunbird/sunbird-devops/tree/release-5.2.0-knowlg/ansible/roles/es-mapping/files

**Reply by krishnakr6** [reply]
@aimansharief ok let me check. Thanks

**Reply by krishnakr6** [reply]
@aimansharief search health check API is working.

Thank you.
